src/CMakeLists.txt

```diff
@@ -50,9 +50,22 @@ endif()
 if(ENABLE_TYPE_DISCOVERY)
   set(DDS_HAS_TYPE_DISCOVERY "1")
 endif()
+
+set(CYCLONE_BUILD_WITH_ICEORYX FALSE CACHE BOOL "Iceoryx not found by default" FORCE)
 if(ENABLE_SHM)
-  set(DDS_HAS_SHM "1")
+  if(CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME MATCHES Darwin)
+    message("Linux/MacOS system found, trying to find iceoryx...")
+    find_package(iceoryx_binding_c QUIET)
+    if(iceoryx_binding_c_FOUND)
+      set(CYCLONE_BUILD_WITH_ICEORYX TRUE CACHE BOOL "Iceoryx found!" FORCE)
+      set(DDS_HAS_SHM "1")
+      message("Build CycloneDDS with iceoryx shared memory support!")
+    else()
+      message("iceoryx not found, falling back to loopback communication.")
+    endif()
+  endif()
```

Comment:
The idea behind is that we try to build with shm support per default and look if iceoryx is present or not. If this is the case we link against it, otherwise it falls back to the loopback communication. This was discussed with @eboasson to ensure always a build on the buildfarm.

Reply:
Maybe a tri-state is neatest (for security as well), but we can change it later. Having this work in is more important short-term.